src/Nri/Ui/ClickableText/V1.elm

```diff
@@ -0,0 +1,163 @@
+-}
+type Size
+    = Small
+    | Medium
+    | Large
+
+
+{-| Config for the button
+-}
+type alias ButtonConfig msg =
+    { label : String
+    , size : Size
+    , icon : Maybe IconType
+    , onClick : msg
+    }
+
+
+{-| Creates a `<button>` element
+-}
+button : ButtonConfig msg -> Html msg
+button config =
+    let
+        fontSize =
+            sizeToPx config.size
+    in
+    Nri.Ui.styled Html.button
+        (dataDescriptor "button")
+        (clickableTextStyles fontSize)
+        [ Events.onClick config.onClick
+        ]
+        [ icon fontSize config.icon
+        , text config.label
+        ]
+
+
+{-| Config for the link
+-}
+type alias LinkConfig =
+    { label : String
+    , size : Size
+    , icon : Maybe IconType
+    , url : String
+    }
+
+
+{-| Creates a `<a>` element
+-}
+link : LinkConfig -> List (Attribute msg) -> Html msg
+link config additionalAttributes =
+    let
+        fontSize =
+            sizeToPx config.size
+    in
+    Nri.Ui.styled Html.a
+        (dataDescriptor "link")
+        (clickableTextStyles fontSize)
+        (Attributes.href config.url
+            :: additionalAttributes
+        )
+        [ icon fontSize config.icon
+        , text config.label
+        ]
+
+
+clickableTextStyles : Css.Px -> List Css.Style
+clickableTextStyles fontSize =
+    [ Css.cursor Css.pointer
+    , -- Specifying the font can and should go away after bootstrap is removed from application.css
+      Nri.Ui.Fonts.V1.baseFont
+    , Css.backgroundImage Css.none
+    , Css.textShadow Css.none
+    , Css.property "transition" "background-color 0.2s, color 0.2s, box-shadow 0.2s, border 0.2s, border-width 0s"
+    , Css.boxShadow Css.none
+    , Css.border Css.zero
+    , Css.disabled [ Css.cursor Css.notAllowed ]
+    , Css.color Colors.azure
+    , Css.backgroundColor Css.transparent
+    , Css.fontWeight (Css.int 600)
+    , Css.textAlign Css.left
+    , Css.borderStyle Css.none
+    , Css.textDecoration Css.none
+    , Css.hover [ Css.textDecoration Css.underline ]
+    , Css.fontSize fontSize
+    , Css.padding Css.zero
+    ]
+
+
+icon : Css.Px -> Maybe IconType -> Html msg
+icon fontSize maybeIcon =
+    case maybeIcon of
+        Just iconType ->
+            -- TODO: We should never use an image here, only SVG
+            Nri.Ui.styled Html.span
+                (dataDescriptor "icon-holder")
+                [ Css.height fontSize
+                , Css.width fontSize
+                , Css.display Css.inlineBlock
+                , Css.marginRight (Css.px 5)
+                ]
+                []
+                [ Icon.decorativeIcon iconType ]
+
+        Nothing ->
+            text ""
+
+
+sizeToPx : Size -> Css.Px
+sizeToPx size =
+    case size of
+        Small ->
+            Css.px 15
+
+        Medium ->
+            Css.px 17
+
+        Large ->
+            Css.px 20
+
+
+dataDescriptor : String -> String
+dataDescriptor descriptor =
+    "borderless-button-v1-" ++ descriptor
```

Comment:
🎨 should we change this to "clickable-text-v1"?

Reply:
Oops yes, I missed this!